packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -69,222 +113,333 @@ abstract class FlutterGoldenFileComparator extends GoldenFileComparator {
+  /// The [fs] and [platform] parameters are useful in tests, where the default
+  /// file system and platform can be replaced by mock instances.
+  FlutterLocalFileComparator(
+    final Uri basedir,
+    final SkiaGoldClient skiaClient, {
+    final FileSystem fs = const LocalFileSystem(),
+    final Platform platform = const LocalPlatform(),
+  }) : super(
+    basedir,
+    skiaClient,
+    fs: fs,
+    platform: platform,
+  );
+
+  /// Creates a new [FlutterLocalFileComparator] that mirrors the
+  /// relative path resolution of the default [goldenFileComparator].
+  ///
+  /// The [goldens] and [defaultComparator] parameters are visible for testing
+  /// purposes only.
+  static Future<FlutterGoldenFileComparator> fromDefaultComparator(
+    final Platform platform, {
+    SkiaGoldClient goldens,
     LocalFileComparator defaultComparator,
-  }) {
+  }) async {
+
     defaultComparator ??= goldenFileComparator;
-    return FlutterSkippingGoldenFileComparator(defaultComparator.basedir);
+    final Directory baseDirectory = FlutterGoldenFileComparator.getBaseDirectory(
+      defaultComparator,
+      platform,
+    );
+
+    if(!baseDirectory.existsSync()) {
+      baseDirectory.createSync(recursive: true);
+    }
+
+    goldens ??= SkiaGoldClient(baseDirectory);
+    await goldens.getExpectations();
+
+    return FlutterLocalFileComparator(baseDirectory.uri, goldens);
   }
 
   @override
   Future<bool> compare(Uint8List imageBytes, Uri golden) async {
-    print('Skipping "$golden" test : Skia Gold is not available in this testing '
-      'environment and flutter/goldens repository comparison is only available '
-      'on Linux machines.'
-    );
-    return true;
-  }
+    golden = _addPrefix(golden);
+    final String testName = skiaClient.cleanTestName(golden.path);
+    final List<String> testExpectations = skiaClient.expectations[testName];
+    if (testExpectations == null) {
+      // There is no baseline for this test
+      print('No expectations provided by Skia Gold for test: $golden. '
+        'This may be a new test. If this is an unexpected result, check '
+        'https://flutter-gold.skia.org.\n'
+        'Validate image output found at $basedir'
+      );
+      update(golden, imageBytes);
+      return true;
+    }
 
-  @override
-  Future<void> update(Uri golden, Uint8List imageBytes) => null;
+    ComparisonResult result;
+    final Map<String, ComparisonResult> failureDiffs = <String, ComparisonResult>{};
+    for (String expectation in testExpectations) {
+      final List<int> goldenBytes = await skiaClient.getImageBytes(expectation);
+
+      result = GoldenFileComparator.compareLists(
+        imageBytes,
+        goldenBytes,
+      );
+
+      if (result.passed) {
+        return true;
+      }
+      failureDiffs[expectation] = result;
+    }
+    failureDiffs.forEach((String expectation, ComparisonResult result) async {
+      if (await skiaClient.isValidDigestForExpectation(expectation, golden.path))
```

Comment:
Waiting to make this call until there is **no** positive found will save **a lot** of calls, as it will only reach this point when a change has been detected.